### ISSUE TITLE
Wrong LD_RUNPATH_SEARCH_PATHS for app extensions's xcconfigs

### ISSUE THREAD

**Issue by vfn** [CocoaPods/CocoaPods#6006]
🌈
# Report
## What did you do?

I added an extension to an app and set it to use my own `.xcconfig` files, with the proper imports the ones generated by Cocoapods.

In order to make the target to use the values set in the `.xcconfig` files, I deleted the values overridden by Xcode in the `Build Settings`

When building and running the app on a device the extension would crash as it could not find the frameworks inside the app's bundle
## What did you expect to happen?

When you create an app extension, Xcode will set `LD_RUNPATH_SEARCH_PATHS` the to

```
LD_RUNPATH_SEARCH_PATHS = $(inherited) @executable_path/Frameworks @executable_path/../../Frameworks
```

<img width="1250" alt="screen shot 2016-10-07 at 1 24 24 pm" src="https://cloud.githubusercontent.com/assets/308592/19177089/aaae4cfe-8c91-11e6-9cd5-20648dac888f.png">

When removing the overridden values you'd expect that Cocoapods generated `.xcconfig` files would have the correct paths
## What happened instead?

Cocoapods doesn't add the correct path for frameworks. The `xcconfig` file contains the line:

```
LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
```

This causes the extension to crash as it cannot find the frameworks inside the app's bundle
### Stack

```
   CocoaPods : 1.1.0.rc.2
        Ruby : ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin15]
    RubyGems : 2.5.0
        Host : Mac OS X 10.12 (16A323)
       Xcode : 8.0 (8A218a)
```

This issue is at some extent related to https://github.com/CocoaPods/CocoaPods/issues/4383 and https://github.com/CocoaPods/CocoaPods/issues/4203. These issues are resolved on `1.1.0.rc.2`, the only missing part of it is the fixing of the `LD_RUNPATH_SEARCH_PATHS` otherwise the `.appex` targets won't be able to find the frameworks from within the app's bundle.

This issue only happens when removing the values overridden by Xcode.

**Comment by orta** [comment]
We've just released 1.1.0.rc.3 with fixes around extensions, can you please update and re-run?

**Comment by vfn** [comment]
@orta I'll have a look at it as soon. Although, the place where the paths are set doesn't look to have changed and does not contain the correct paths for app extensions:

https://github.com/benasher44/CocoaPods/blob/master/lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb#L216
```Ruby
        def generate_ld_runpath_search_paths
          ld_runpath_search_paths = ['$(inherited)']
          if target.platform.symbolic_name == :osx
            ld_runpath_search_paths << "'@executable_path/../Frameworks'"
            ld_runpath_search_paths << \
              if target.native_target.symbol_type == :unit_test_bundle
                "'@loader_path/../Frameworks'"
              else
                "'@loader_path/Frameworks'"
              end
          else
            ld_runpath_search_paths << [
              "'@executable_path/Frameworks'",
              "'@loader_path/Frameworks'",
            ]
          end
          @xcconfig.merge!('LD_RUNPATH_SEARCH_PATHS' => ld_runpath_search_paths.join(' '))
        end
```

**Comment by orta** [comment]
@benasher44 ^

**Comment by benasher44** [comment]
I wouldn't remove the overridden values for your target. Last I checked, Xcode has good default values there that make sense base on the target you're using. Extensions for example have slightly different ones conpared to application targets.

That said, it does look like we try to set some up for you. It should be easy to update this for extension targets. I might try to figure out if we can remove this from CocoaPods though, since Xcode's defaults should "just work," and it's otherwise an added maintenance burden.

**Comment by benasher44** [comment]
I suspect though that we can't really remove this functionality from CP, since people likely rely on this now

**Comment by benasher44** [comment]
For those looking for a workaround here, you can add these values to your "Runpath Search Paths" settings for your extension targets in Xcode: "$(inherited) @executable_path/../../Frameworks". However, extension-related issues seem to have stabilized for this release (1.1.0), so I don't want to want to introduce any changes in that area at this point. This is on my radar to fix in an upcoming release, but I don't plan to fix it for 1.1.0.

**Comment by benasher44** [comment]
The fix here is to check `target.requires_host_target?` in `generate_ld_runpath_search_paths` in `lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb`, and add `@executable_path/../../Frameworks` if that check passes.
